src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -2308,15 +2293,15 @@ private static unsafe void ThreadSafeCopy(char* sourcePtr, char[] destination, i
         {
```

Comment:
Can we replace the whole thing with just:
```
private static unsafe void ThreadSafeCopy(char* sourcePtr, char[] destination, int destinationIndex, int count) =>
    new ReadOnlySpan(sourcePtr, count).CopyTo(new Span(destination, destinationIndex, count));
```

The exception thrown if there is a race condition that corrupts the StringBuilder may be slightly different, but I do not think it matters.

We had *many* subtle bugs in the StringBuilder manual bounds checking in the past. It would be really nice to get rid of the manual checks in StringBuilder.